### PR TITLE
feat: [PIE-5229]: async pipeline validation

### PR DESCRIPTION
**Pull request by rameshx**
### Summary

Adds initial changes for rendering different validation badges and relevant dialogs in pipeline studio that show validation status that is fetched asynchronously and allow revalidation when required.

New feature flag: PIE_ASYNC_VALIDATION

#### Screenshots

https://user-images.githubusercontent.com/52284933/222407190-f4583aa5-5acc-4648-84b4-feb9b5a46f22.mov

### CODE REVIEW

File: src/framework/strings/__tests__/Strings.test.tsx [hunk 1]
```diff
@@ -156,12 +156,12 @@ describe('useString tests', () => {
...
       const { result, rerender } = renderHook(useStrings, { wrapper })
-      const prevGetString = result.current
 
+      const prevGetString = result.current.getString
       rerender()
-      const nextGetString = result.current
+      const nextGetString = result.current.getString
 
-      expect(prevGetString).toEqual(nextGetString)
+      expect(prevGetString).toBe(nextGetString)
```
Comment (rameshx) [inline src/framework/strings/__tests__/Strings.test.tsx:164]: unrelated to the rest of the PR, just an improvement of recently added test

File: src/modules/10-common/strings/strings.en.yaml [hunk 2]
```diff
@@ -1356,6 +1356,8 @@ python: Python
 seeAdvancedOptions: See Advanced Options
 addedEntityLabel: 'Added {{entity}}'
 bucketName: Bucket Name
+validationInProgress: Validation in progress
```
Comment (rameshx) [inline src/modules/10-common/strings/strings.en.yaml:1359]: this is added in common as the same string is used in multiple modules

File: src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationBadge.tsx [hunk 3]
```diff
@@ -0,0 +1,221 @@
...
+  const policyEval = validationResultData?.data?.policyEval as Evaluation | undefined
+  const endTs = validationResultData?.data?.endTs
+
+  const shouldPoll = (): boolean => {
+    return !!validationUuid && !validationResultLoading && isStatusLoading(status)
+  }
+
+  usePolling(refetchValidationResult, {
+    pollingInterval: 5_000,
+    startPolling: !!validationUuid && !validationResultLoading && shouldPoll()
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationBadge.tsx:100]: wouldn't `shouldPoll()` be enough?

File: src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationPopoverContent.tsx [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+//         {getString('common.validationInProgress')}
+//       </Text>
+//       <Text font={{ variation: FontVariation.BODY }} color={Color.GREY_100}>
+//         {getString('pipeline.validation.validationInProgressText')}
+//       </Text>
+//     </div>
+//   )
+// }
+
+export function ErrorPopoverContent({ errorCount = 1 } = {}): JSX.Element {
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationPopoverContent.tsx:58]: ValidationSuccessPopoverContent and ErrorPopoverContent are almost similar. Maybe, we can pass diff props and keep this as just one component.

**Review by chetanvekariya-harness — DISMISSED** [review]

**Review by vkbansal-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationPopoverContent.tsx [hunk 5]
```diff
@@ -9,31 +9,47 @@ import React from 'react'
...
+
   return (
     <div className={css.popoverContent}>
       <Text
-        icon="tick"
-        iconProps={{ color: Color.PRIMARY_4 }}
         font={{ variation: FontVariation.BODY2 }}
         color={Color.GREY_0}
         margin={{ bottom: 'xsmall' }}
+        {...(isError && { icon: 'warning-sign', iconProps: { color: Color.RED_700 } })}
```
Comment (chetanvekariya-harness) [inline src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationPopoverContent.tsx:38]: We can handle this in next PR or some time later. But, what I meant is you can pass label, icon, title etc. as props and just use it here. No conditions. 
Basically, you call this component twice with different props.

**Review by chetanvekariya-harness — APPROVED** [review]

File: src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationPopoverContent.tsx (hunk 5 above)
Comment (rameshx) [reply src/modules/70-pipeline/components/PipelineStudio/AsyncValidation/ValidationPopoverContent.tsx:38]: Sure. I have to change this anyway when I enable the loading popover.

**Pull request merged**